src/app/shared/search-box/query-item/query-item.component.html

```diff
@@ -1,5 +1,5 @@
 <div [style.background-color]="queryItem | queryItemBackground"
-     [style.border-color]="queryItem | queryItemBorder"
+     [style.border-color]="queryItem | queryItemBorder:form.valid"
```

Comment:
I think you need to check if form control is valid and not the whole form. I think all query items will be red now even the valid ones.

Reply:
It's not the whole form. I'm passing `[form]="queryItemsControl.at(itemIndex)"` into `query-item.component.html` and everything works as expected.

Reply:
Oh, then it is ok. Just the name is confusing.